File: php/src/framework/FrameworkResponse.php
<?php
declare(strict_types=1);

namespace Demo\App\framework;

final readonly class FrameworkResponse
{
    public const STATUS_OK = 200;
    public const STATUS_CREATED = 201;
    public const STATUS_BAD_REQUEST = 400;

    public function __construct(private int $statusCode, private array $data)
    {
    }

    public function statusCode(): int
    {
        return $this->statusCode;
    }

    public function data(): array
    {
        return $this->data;
    }
}

File: kotlin/src/main/kotlin/framework/FrameworkResponse.kt
package framework

data class FrameworkResponse(val statusCode: Int, val content: Map<String, String>)
{
    companion object {
        const val STATUS_OK = 200
        const val STATUS_CREATED = 201
        const val STATUS_BAD_REQUEST = 400
        const val STATUS_NOT_FOUND = 404
    }
}
